cloudify_azure/resources/compute/virtualmachine.py

```diff
@@ -225,8 +226,8 @@ def _handle_userdata(existing_userdata):
     elif isinstance(existing_userdata, dict) or \
             isinstance(existing_userdata, list):
         existing_userdata = json.dumps(existing_userdata)
-    elif not isinstance(existing_userdata, basestring):
-        existing_userdata = str(existing_userdata)
+    elif not isinstance(existing_userdata, text_type):
```

Comment:
I think this is will change behaviour of this code in PY2.  Before this patch we compared type of `existing_userdata` with `basestring`, after – with `unicode`.